### ISSUE TITLE
[Feature request] Improve blib magic to have an option to report relative to lib/ 

### ISSUE THREAD

**Issue by kentfredric** [pjcj/Devel--Cover#91]
There's an annoying problem I've faced using coveralls + travis, and that is, if you build your code in such a way `blib/` exists, the auto-blib trick loads `blib/` instead, and so the resulting report that gets sent to coveralls contains only references to `blib/`, which coveralls can't find in the respective git commit.

For the mean time the only hack I have to solve this issue is avoiding the existence of `blib/` entirely. That's fine for me, because fortunately, I don't have any XS modules or anything of that sort, and no magic `lib/ -> blib` transformations.

In short, it would be nice if there was a way to report coverage relative to source files ie:

```
lib/Foo.pm     4  4 4 5 
```

Instead of 

```
blib/lib/Foo.pm 4 4 4 5
```

**even** if blib was indeed used.

One way it might be feasible is if you can assume things about transformation, and report from `lib/` only if `blib/$PATH` and `lib/$PATH` match at the line of the covered data, and report only `blib/$PATH` when `path("blib/$PATH")->lines()[$i] ne path("$PATH")->lines()[$i]`

**Comment by pjcj** [comment]
Thanks very much for this report.  Should I understand that the workaround above is sufficient and no changes are needed in Devel::Cover, or would you still prefer another solution within Devel::Cover?

(Sorry for taking so long to get back to you on this one.)

**Comment by kentfredric** [comment]
> or would you still prefer another solution within Devel::Cover?

Would prefer an option still, I think. That time delay makes it rather difficult to reconnect with my present mindset :) 

At very least these sorts of situations should be documented sufficient to give people in similar situations a clear path forward.

**Comment by pjcj** [comment]
I have added the -prefer_lib option to cover with 089eb668d39fca39e44a51bbeed0f6e4d61a8cfb to prefer the file in the lib directory even if the file in blib was used.  Thanks a lot for this report.
